lnd.go

```diff
@@ -438,6 +440,10 @@ func Main(cfg *Config, lisCfg ListenerCfg, shutdownChan <-chan struct{}) error {
 		}
 	}
 
+	// Now we're definitely done with the unlocker, shut it down so we can
+	// start the main RPC service later.
+	shutdownUnlocker()
```

Comment:
In later commits, when we send the admin macaroon back to the `wallet unlocker service`, can we really know at this point that we can shut it down? I'm thinking the moment the service gets the macaroon, it looks like we don't wait for it to be returned to the caller before calling this shutdown method.

Reply:
Yeah, you're right. I added a wait group to catch this.